### ISSUE TITLE
Force youtube quality or speed

### ISSUE THREAD

**Issue by basilelt** [dmunozv04/iSponsorBlockTV#200]
Would it be possible to add a way in iSponsorBlockTV to send a forced resolution (like 4k60) and even a force speed control (like 1.25).

It would be a really nice feature as YouTube really likes to set it to auto quality or even 480p.

And for speed it gets reset pretty often.

Thank you in advance !

**Comment by basilelt** [comment]
Wanted to take a look myself but : https://stackoverflow.com/questions/37846208/how-to-set-playback-quality-youtube-api

I know a similar project that use another way to control youtube to skip sponsor : https://github.com/bsharper/sbremote
And uses Apple TV remote controls from network pairing, maybe something similar could be possibly made to change quality. I changed to your solution because it's more versatile and the previously mentioned doesn't work anymore (maybe apple/youtube changed some stuff).

**Comment by dmunozv04** [comment]
It would certainly be possible and a nice to have, considering YouTube last week decided to change the quality on my TV to 720p again. 
This project doesn’t use the regular YouTube api, but rather their internal (and undocumented) lounge api; the one the YouTube app on your phone uses to remote control the YouTube app on a TV. 

If you want to dive into it, I’d suggest checking [pyytlounge](https://github.com/FabioGNR/pyytlounge?tab=readme-ov-file), the wonderful library this project uses; and [this wiki ](https://github.com/enriquecidok/youtube-lounge-api/wiki)that describes a bunch of things about the API. 
Another way to find new events like changing resolution or playback speed is to setup a recording proxy like Charles proxy or Proxyman on your phone, record the traffic when changing that setting and identifying what’s happening. 
Do let me know if you need anything else, I hope you find this useful

**Comment by basilelt** [comment]
It seems like either YouTube is not allowing remote changing of the quality through the lounge api or they haven't implemented it yet.
![Screenshot_20241116-233113.png](https://github.com/user-attachments/assets/d30d5459-0b02-44d3-8021-684330fb8f8f)

**Comment by basilelt** [comment]
I tried both normal and beta versions of youtube on Android, no luck sadly.

**Comment by dmunozv04** [comment]
Yeah, you're right; it used to work before but it's broken currently

**Comment by sternma** [comment]
I looked into this a bit; unfortunately, it won't be possible to adjust the video quality via the Lounge API until YouTube provides that functionality from their app too.

On the other hand, I submitted a PR upstream to allow this package to change playback speed: https://github.com/FabioGNR/pyytlounge/pull/14
